core/src/com/unciv/logic/automation/UnitAutomation.kt

```diff
@@ -397,6 +399,28 @@ class SpecificUnitAutomation{
         else UnitAutomation().explore(unit, unit.getDistanceToTiles())
     }
 
+    fun automateGeneral(unit: MapUnit){
+        //try to follow nearby units. Do not garrison in city if possible
+        val militantToCompany = unit.civInfo.getCivUnits()
+                .firstOrNull { val tile = it.currentTile
+                    it.type.isLandUnit() && it.getMaxMovement() <= 2.0f && tile.civilianUnit==null
+                            && unit.canMoveTo(tile) && unit.movementAlgs().canReach(tile) && !tile.isCityCenter() }
```

Comment:
canReach() should always be the last check in any set of boolean checks, since it's the most CPU and time intensive

Reply:
Agree.